presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -633,6 +633,50 @@ TEST_F(TaskManagerTest, tableScanAllSplitsAtOnce) {
   assertResults(taskId, rowType_, "SELECT * FROM tmp WHERE c0 % 5 = 0");
 }
 
+TEST_F(TaskManagerTest, fecthFromFinishedTask) {
+  auto filePaths = makeFilePaths(5);
+  auto vectors = makeVectors(filePaths.size(), 1'000);
+  for (int i = 0; i < filePaths.size(); i++) {
+    writeToFile(filePaths[i]->path, vectors[i]);
+  }
+  duckDbQueryRunner_.createTable("tmp", vectors);
+
+  auto planFragment = exec::test::PlanBuilder()
+                          .tableScan(rowType_)
+                          .filter("c0 % 5 = 0")
+                          .partitionedOutputArbitrary({"c0", "c1"})
+                          .planFragment();
+
+  long splitSequenceId{0};
+
+  protocol::TaskId taskId = "scan.0.0.1.0";
+
+  protocol::TaskUpdateRequest updateRequest;
+  updateRequest.sources.push_back(
+      makeSource("0", filePaths, true, splitSequenceId));
+  auto taskInfo = createOrUpdateTask(taskId, updateRequest, planFragment);
+
+  auto shortWait = std::chrono::seconds(1);
+  auto longWait = protocol::Duration("300s");
+  auto maxSize = protocol::DataSize("32MB");
+  auto resultRequestState = http::CallbackRequestHandlerState::create();
+  auto eventBase = folly::EventBaseManager::get()->getEventBase();
+
+  // Wait for drivers to finish processing data
```

Comment:
You can wait for the task state transition here?

Reply:
when we call `taskManager_->abortResults(taskId, destination);` the task can finish only when
1. all splits have been processed successfully
2. all destinations have been deleted

so at first we let the split to finish processing and then call `taskManager_->abortResults(taskId, destination);`

Reply:
Yes, so we can wait for the task state transition instead of having a sleep here? We shall avoid such timing flakiness in test if possible.